### ISSUE TITLE
Update route constants

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/mobile#2512]
## Is your feature request related to a problem? Please describe.

Placeholder issue for updating `getColumns` with route constants.

## Describe the solution you'd like

See above.

## Implementation

Update navigation constants, add to `getColumns`.

## Describe alternatives you've considered

N/A.

## Additional context

N/A.

**Comment by joshxg** [comment]
@wlthomson 

doesn't seem to urgent - especially not in to a patch? Is the milestone a mistake?

**Comment by wlthomson** [comment]
@joshxg My thinking was to keep `v5.0.0` relatively limited to functional changes and to keep bug fixes and refactors/tidy ups to patches. Patches are ideally going to be on a rolling basis, so non-urgency doesn't mean it's more suitable to a major bump IMO. But don't think it really makes a difference either way, so have bumped!

**Comment by joshxg** [comment]
@wlthomson 

Ah ok - I suppose with the overhead of releasing a new version without having automated systems in place (for releasing and testing) I assumed (maybe wrongly) that patches were for relatively urgent bug fixes. I also thought that in general patch version bumps with semantic versioning were for bug fixes specifically.

I also thought that the only difference between major and minor bumping is that a major is backwards incompatible - the type of content (bug fix, feature or refactor) didn't matter?

**Comment by wlthomson** [comment]
@joshxg I think the overhead of a new patch version is relatively low? I agree that it would be more suitable to a minor release, but I think logistically at this stage rolling patches is easier/more efficient.

You're right re patches being for bug fixes, but I don't think limited only to that. Generally, semantic versioning guidelines don't really talk about refactoring (https://semver.org/), but the emphasis on major and minor is on functionality (e.g. major for backwards incompatible, minor otherwise), so think patches are probably the best place for (minor) refactors as well as bug fixes. That way the minor and majors stay relatively uncluttered and the changelogs are more meaningful (IMO someone reading a major release changelog is probably more interested in the API changes than any internal refactors). 

TLDR; I think minor refactors are most suited to patches and keep minor and major to new/changed functionality.

**Comment by joshxg** [comment]
So this is probably not the best place for the discussion, but why not aye?! 🤣 

> I think the overhead of a new patch version is relatively low?

But there is an overhead. And not that low when considering that we don't have any CI/CD, automatic testing, proper app updating or even a proper process around applying new versions for users

> I agree that bigger refactors would be more suitable to a minor release

I don't see the difference in a minor or major refactor. There is internal code changes which can cause regressions (especially when lacking automated tests) and offer no benefit to the end-user. Why release a version which does nothing for the end-user?

> but I think logistically at this stage rolling patches is easier/more efficient.

I don't see how it is easier to:

- branch from master
- make a PR into the master branch with potentially less testing (the code is in the code-base for less time before making it to the end-user, so there is no day-to-day testing during development of other features)
- merge that back through any branches from master
- Release a version

V

- branch from develop
- merge into develop

All this (ok, it's exaggerated and it's not that much but it is something) with no benefit to the end-user. 

What are the pros for putting refactors into a patch?

> You're right re patches being for bug fixes, but I don't think limited only to that.

Sure - but what are the benefits of putting minor refactors into a patch, I only see un-needed overhead

> That way the minor and majors stay relatively uncluttered and the changelogs are more meaningful (IMO someone reading a major release changelog is probably more interested in the API changes than any internal refactors).

If this is the only benefit, I don't see it being very valid - "Internal changes made to X" is hardly so intrusive that we should change process' for it? More than that, you now have potentially more patches which you have to look through when trying to find specific changes in changelogs? Again, not a big deal, but just as much of a negative as your point re the changelog, imo.

> TLDR; I think minor refactors are most suited to patches and keep minor and major to new/changed functionality.

I just don't see any benefit to putting refactors into patches over a release

**Comment by wlthomson** [comment]
> But there is an overhead. And not that low when considering that we don't have any CI/CD, automatic testing, proper app updating or even a proper process around applying new versions for users

Yeah but we're not talking about releasing patches with only refactors? We're discussing whether we should be putting refactors into patches or minor/major? IMO, the overhead of small patches is actually a lot less than delaying to a big releases, as the regression testing can be done more agile and continuously, rather than in big "test pushes" which IMO are more inefficient.  

Also, I think big factor here is that that releasing a patch doesn't mean releasing to a user, it should, but in practice without CI and without clients which consistent internet etc. it's not that easy. I think the discussion is more about whether we should have regular small releases rather than infrequent big ones, and IMO the benefits outweigh the negatives.

> I don't see the difference in a minor or major refactor. There is internal code changes which can cause regressions (especially when lacking automated tests) and offer no benefit to the end-user. Why release a version which does nothing for the end-user?

See above. IMO big changes = more surface for regressions = more testing. Small refactors can be more easily tested, and by keeping testing continuously moving forward, we can more easily organise big refactors when there is more man power for regression testing.

>I don't see how it is easier to:
>
>    branch from master
>    make a PR into the master branch with potentially less testing (the code is in the code-base for less time before making it to the end-user, so there is no day-to-day testing during development of other features)
>    merge that back through any branches from master
>    Release a version
>    branch from develop
>    merge into develop
> All this (ok, it's exaggerated and it's not that much but it is something) with no benefit to the end-user.

Yeah, I think all up that's a pretty small overhead, and isn't really that much more than doing a regular PR! Maybe 15 mins? IMO the overhead isn't really an issue and I see a decent benefit to our ability to keep things organised and split the workload of big releases into smaller chunks. Really, the discussion is do we want regular small releases or less frequent big ones. I think the general industry concensus is that smaller, flexible release cycles are better than big infrequent ones.

> Sure - but what are the benefits of putting minor refactors into a patch, I only see un-needed >overhead

As above, testing etc. is a lot easier to manage and keeps us moving towards a more agile model which (IMO) is the way to go.

>If this is the only benefit, I don't see it being very valid - "Internal changes made to X" is hardly >so intrusive that we should change process' for it? More than that, you now have potentially >more patches which you have to look through when trying to find specific changes in >changelogs? Again, not a big deal, but just as much of a negative as your point re the >changelog, imo.

I don't think it's changing process? My impression was that we wanted to move towards doing things more agile, smaller cycles etc. Also a common changelog format is to group the lower releases into the bigger release notes, e.g. the patch changes are grouped into the changes for the next minor (and so on). IMO this doesn't take away from your ability to find things, it just makes them in smaller chunks and easier to manage.

> I just don't see any benefit to putting refactors into patches over a release

I think it just comes down to whether you prefer fewer bigger releases or smaller releases. IMO, industry generally goes with the latter, hence the popularity of agile, SCRUM, sprints etc. I feel the latter is preferable for the reasons above, but it's not a hill I'm going to die on!

**Comment by wlthomson** [comment]
@joshxg My previous reply doesn't really offer a solution, so how about something like this:

- Patches: regular bug fixes, released roughly weekly.
- Minors: refactors and backwards compatible functionality, released roughly fortnightly.
- Majors: backwards incompatible functionality, released roughly monthly.

By attaching rough time outlines, it becomes a bit less arbitrary and a bit more "agile". I also think by making minors more regular, pushing refactors into them would be appropriate, as the main reason I wanted them in patches is due to the flexibiliy of the release cycle for shorter cycles.

Thoughts?

**Comment by wlthomson** [comment]
Fixed by @2514.
